### PR TITLE
fix wrong deltat coefficient

### PR DESCRIPTION
**Pull request by cedricleroy**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python! You may delete all of these instructions except for the list below.

You may submit a pull request with your code at any stage of completion.

The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items below:

 - [ ] Closes issue #xxxx
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

* Per the [reference](https://eclipse.gsfc.nasa.gov/SEcat5/deltatpoly.html) cited, the constant coefficient of the polynomial expression for years >= 1860 and < 1900 should be 7.62 instead of 7.6.
* Year 2150 was not covered.

### CODE REVIEW

File: pvlib/test/test_spa.py [hunk 1]
```diff
@@ -79,13 +79,15 @@
 Phi = 194.340241
 year = 1985 
 month = 2
-year_array = np.array([-499, 500, 1000, 1500, 1800, 1900, 1950, 1970, 1985, 1990, 2000, 2005])
-month_array = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
+year_array = np.array([-499, 500, 1000, 1500, 1800, 1860, 1900, 1950,
+                       1970, 1985, 1990, 2000, 2005])
```
Comment (wholmgren) [inline pvlib/test/test_spa.py]: perhaps we should add 2050 and 2150 to this test array

File: pvlib/test/test_spa.py [hunk 2]
```diff
@@ -79,13 +79,15 @@
...
+month_array = np.array([1, 2, 3, 4, 5, 6, 6, 7, 8, 9, 10, 11, 12])
```
Comment (wholmgren) [inline pvlib/test/test_spa.py]: two `6`s?

Comment (mikofski) [reply pvlib/test/test_spa.py]: why not?
```python
np.arange(1,13)
# array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12])
```

Comment (cedricleroy) [reply pvlib/test/test_spa.py]: Given how the tests are currently designed, `year_array` and `month_array` need to have the same number of elements. The test covering the years is doing pairs with `month_array`.
Adding one year to test lead to an additional month, that's why we have two 6 (could be something else). If we add 2050 and 2150, we will need two additional months. Or, we need to revamp the tests.

Comment (wholmgren) [reply pvlib/test/test_spa.py]: Ah, thanks. I suggest we add the extra months to the array and a comment explaining the duplicates.

File: pvlib/spa.py [hunk 3]
```diff
@@ -1418,7 +1418,7 @@ def calculate_deltat(year, month):
                       -20+32*((y-1820)/100)**2
                       - 0.5628*(2150-y), deltat)
 
-    deltat = np.where(year > 2150,
+    deltat = np.where(year >= 2150,
```
Comment (markcampanelli) [inline pvlib/spa.py]: @cedricleroy This doesn't appear to solve an apparent discontinuity issue at year 2150. I get the following with and without this change:
```console
inputs_year, inputs_month = np.meshgrid(np.array([2149.999999999999, 2150, 2150.000000000001]), np.arange(1, 13))
calculate_deltat(inputs_year, inputs_month)
----
array([[ 328.59145556,    0.        ,  328.56800556],
       [ 328.8144    ,    0.        ,  328.74405   ],
       [ 329.03738889,    0.        ,  328.92013889],
       [ 329.26042222,    0.        ,  329.09627222],
       [ 329.4835    ,    0.        ,  329.27245   ],
       [ 329.70662222,    0.        ,  329.44867222],
       [ 329.92978889,    0.        ,  329.62493889],
       [ 330.153     ,    0.        ,  329.80125   ],
       [ 330.37625556,    0.        ,  329.97760556],
       [ 330.59955556,    0.        ,  330.15400556],
       [ 330.8229    ,    0.        ,  330.33045   ],
       [ 331.04628889,    0.        ,  330.50693889]])
```

**Comment by wholmgren** [comment]
@cedricleroy sorry I missed the updates a few weeks ago. 

@markcampanelli thanks for the review.

File: pvlib/spa.py (hunk 3 above)
Comment (cedricleroy) [reply pvlib/spa.py]: Possible to check again @markcampanelli?
Here is what I get with 5ee82b1:
```python
from pvlib.spa import calculate_deltat
>>> calculate_deltat(2150, 6)
0.0
```
With 9aee4fc:
```python
from pvlib.spa import calculate_deltat
>>> calculate_deltat(2150, 6)
329.4486722222225
```

Comment (markcampanelli) [reply pvlib/spa.py]: Sorry if I messed this up. I will check again and report back.

Comment (markcampanelli) [reply pvlib/spa.py]: Indeed the conda environment I was using wasn't grabbing the checkout, and I screwed up the year between checkouts in my first check which made me think I actually was using different code in Jupyter. Everything looks good:
```python
inputs_year, inputs_month = np.meshgrid(np.array([2149.999999999999, 2150, 2150.000000000001]), np.arange(1, 13))
calculate_deltat(inputs_year, inputs_month)
```
```python
array([[328.59145556, 328.56800556, 328.56800556],
       [328.8144    , 328.74405   , 328.74405   ],
       [329.03738889, 328.92013889, 328.92013889],
       [329.26042222, 329.09627222, 329.09627222],
       [329.4835    , 329.27245   , 329.27245   ],
       [329.70662222, 329.44867222, 329.44867222],
       [329.92978889, 329.62493889, 329.62493889],
       [330.153     , 329.80125   , 329.80125   ],
       [330.37625556, 329.97760556, 329.97760556],
       [330.59955556, 330.15400556, 330.15400556],
       [330.8229    , 330.33045   , 330.33045   ],
       [331.04628889, 330.50693889, 330.50693889]])
```

**Comment by wholmgren** [comment]
Thanks!

**Pull request merged**